### PR TITLE
fix(storage): specify credentials with STORAGE_EMULATOR_HOST

### PR DESCRIPTION
**Pull request by shuheiktgw**
Encountered the same problem with https://github.com/googleapis/google-api-go-client/issues/558 when specifying `STORAGE_EMULATOR_HOST`. For our case, we'd like to pass `option.WithTokenSource`, and it raises the following error with `STORAGE_EMULATOR_HOST` during the test. To avoid the error, we should pass `internaloption.SkipDialSettingsValidation` as the Spanner client does.

```
options.WithoutAuthentication is incompatible with any option that provides credentials
```

### CODE REVIEW

File: storage/storage_test.go [hunk 1]
```diff
@@ -1992,6 +1992,20 @@ func TestAttrToFieldMapCoverage(t *testing.T) {
...
 
+func TestEmulatorWithCredentialsFile(t *testing.T) {
+	originalStorageEmulatorHost := os.Getenv("STORAGE_EMULATOR_HOST")
+	defer os.Setenv("STORAGE_EMULATOR_HOST", originalStorageEmulatorHost)
+
+	os.Setenv("STORAGE_EMULATOR_HOST", "localhost:1234")
+
+	client, err := NewClient(context.Background(), option.WithCredentialsFile("/path/to/key.json"))
+	if err != nil {
+		t.Errorf("failed creating a client with credentials file when running agains an emulator: %v", err)
```
Comment (tritone) [inline storage/storage_test.go:2003]: you can just use t.Fatalf()

**Review by tritone — COMMENTED** [review]
Thanks for the PR! I think this looks good to me. @BrennaEpp or @codyoss any thoughts?

File: storage/storage_test.go (hunk 1 above)
Comment (shuheiktgw) [reply storage/storage_test.go]: Thank you. Fixed it 🙂

**Comment by shuheiktgw** [comment]
@tritone @BrennaEpp @codyoss I've fixed the PR, so would you review it again? 🙏

File: storage/storage_test.go [hunk 2]
```diff
@@ -1992,6 +1992,20 @@ func TestAttrToFieldMapCoverage(t *testing.T) {
 	}
 }
 
+func TestEmulatorWithCredentialsFile(t *testing.T) {
+	originalStorageEmulatorHost := os.Getenv("STORAGE_EMULATOR_HOST")
+	defer os.Setenv("STORAGE_EMULATOR_HOST", originalStorageEmulatorHost)
+
+	os.Setenv("STORAGE_EMULATOR_HOST", "localhost:1234")
```
Comment (BrennaEpp) [inline storage/storage_test.go:1999]:
```suggestion
	t.Setenv("STORAGE_EMULATOR_HOST", "localhost:1234")
```

**Review by BrennaEpp — COMMENTED** [review]
Just one suggestion, otherwise looks good.

File: storage/storage_test.go (hunk 2 above)
Comment (shuheiktgw) [reply storage/storage_test.go]: Didn't know t.Setenv, thank you!

**Comment by shuheiktgw** [comment]
@tritone @BrennaEpp @codyoss Thank you for your review! Would you review the PR again?

**Review by tritone — APPROVED** [review]
thanks for the contribution!

**Pull request merged**